File: week-01/day-5/src/DrawChessTable.java
public class DrawChessTable {
    public static void main(String[] args) {
        // Crate a program that draws a chess table like this
        //
        // % % % %
        //  % % % %
        // % % % %
        //  % % % %
        // % % % %
        //  % % % %
        // % % % %
        //  % % % %
        //

        // set a bool variable to keep track of the linestart
        // (one of the countless methods)
        boolean startWithPercent = true;

        // iterate through lines
        for (int line = 0; line < 8; line++) {
            // print a line starting with percent
            if (startWithPercent) {
                for (int j = 0; j < 8; j++) {
                    if (j % 2 == 0) {
                        System.out.print("%");
                    } else {
                        System.out.print(" ");
                    }
                }
            } else {
                for (int j = 0; j < 8; j++) {
                    if (j % 2 == 0) {
                        System.out.print(" ");
                    } else {
                        System.out.print("%");
                    }
                }
            }
            // linebreak
            System.out.println();
            // negate the bool to go to the next line properly
            startWithPercent = !startWithPercent;
        }
    }
}

File: python_solutions/week-01/day-4/draw_chess_table.py
# Crate a program that draws a chess table like this
#
# % % % %
#  % % % %
# % % % %
#  % % % %
# % % % %
#  % % % %
# % % % %
#  % % % %

for i in range(8):
    if i % 2 == 0:
        message = "{1}{0}{1}{0}{1}{0}{1}{0}"
    else:
        message = "{0}{1}{0}{1}{0}{1}{0}{1}"

    print(message.format(" ", "%"))

